mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -400,17 +400,13 @@ class Word2Vec extends Serializable with Logging {
     }
     newSentences.unpersist()
     
-    val word2VecMap = mutable.HashMap.empty[String, Array[Float]]
+    val wordArray = new Array[String](vocabSize)
     var i = 0
     while (i < vocabSize) {
-      val word = bcVocab.value(i).word
-      val vector = new Array[Float](vectorSize)
-      Array.copy(syn0Global, i * vectorSize, vector, 0, vectorSize)
-      word2VecMap += word -> vector
+      wordArray(i) = bcVocab.value(i).word
```

Comment:
This is executing on the driver, so it should not use broadcast variables.  Use ```vocab```  Could be shorter to do:
```
val wordArray = vocab.map(_.word)
```